plugins/sigma/src/plugin.cpp

```diff
@@ -182,11 +184,34 @@ class sigma_operator final : public crtp_operator<sigma_operator> {
   std::string path_ = {};
 };
 
-class plugin final : public virtual operator_plugin<sigma_operator> {
+class plugin final : public virtual operator_plugin<sigma_operator>,
+                     public virtual operator_factory_plugin {
   auto signature() const -> operator_signature override {
     return {.transformation = true};
   }
 
+  auto make(invocation inv, session ctx) const
+    -> failure_or<operator_ptr> override {
+    auto refresh_interval = std::optional<located<duration>>{};
+    auto path = std::string{};
+    argument_parser2::operator_("sigma")
+      .add(path, "<path>")
+      .add("refresh_interval", refresh_interval)
+      .parse(inv, ctx)
+      .ignore();
+    // XXX: Is there a way to get a located default value?
```

Comment:
There's `location::unknown`, so you can do `refresh_interval.value_or({default_duration, location::unknown})`.

Reply:
I think you can also use `inv.get_location()` (not exactly sure), which would give you the entire operator's location.